src/app/public/modules/colorpicker/colorpicker.component.ts

```diff
@@ -150,16 +195,29 @@ export class SkyColorpickerComponent implements OnInit, OnDestroy {
       .subscribe((message: SkyColorpickerMessage) => {
         this.handleIncomingMessages(message);
       });
+
+    this.addTriggerButtonEventListeners();
   }
 
   public ngOnDestroy() {
     this.ngUnsubscribe.next();
     this.ngUnsubscribe.complete();
+    this.removePickerEventListeners();
+    this.destroyAffixer();
+    this.destroyOverlay();
+  }
+
+  public onButtonClick(): void {
+    this.sendMessage(SkyColorpickerMessageType.Open);
   }
 
   public closePicker() {
     this.setColorFromString(this.lastAppliedColor);
-    this.closeDropdown();
+    this.destroyAffixer();
+    this.destroyOverlay();
+    this.removePickerEventListeners();
+    this.triggerButtonRef.nativeElement.focus();
```

Comment:
In your visual test, I noticed closing the picker always focuses on the last trigger button, no matter which picker is open.

Reply:
Good catch.